include/fc/exception/exception.hpp

```diff
@@ -446,11 +446,11 @@ namespace fc
    catch( fc::exception& er ) { \
       FC_RETHROW_EXCEPTION( er, LOG_LEVEL, FORMAT, __VA_ARGS__ ); \
    } catch( const std::exception& e ) {  \
-      fc::exception fce( \
+      throw fc::exception( \
                 FC_LOG_MESSAGE( LOG_LEVEL, "${what}: " FORMAT,__VA_ARGS__("what",e.what())), \
                 fc::std_exception_code,\
                 typeid(e).name(), \
-                e.what() ) ; throw fce;\
+                e.what() ) ;\
```

Comment:
I get a warning on these called "throw by value, catch by reference" -- the compiler recommended throwing an anonymous value if you plan to catch by reference, since a named, stack-allocated value technically can't be caught by reference (the stack it was allocated on has been destroyed).

Compiler handles it correctly either way, but this way is more complicit with formal C++ memory rules.